python/tests/test_c_array.py

```diff
@@ -610,3 +622,91 @@ def test_c_array_duration():
         d2_duration_in_ms,
         d3_duration_in_ms,
     ]
+
+
+def test_device_array_errors():
+    from nanoarrow.device import DeviceType, resolve
+
+    try:
+        cuda_device = resolve(DeviceType.CUDA, 0)
+    except ValueError:
+        pytest.skip("CUDA device not available")
```

Comment:
It could be nice to put this in a pytest.fixture if you want to make it reusable e.g.
```
@pytest.fixture
def cuda_device():
    from nanoarrow.device import resolve
    try:
        return resolve(DeviceType.CUDA, 0)
    except ValueError:
        pytest.skip("CUDA device not available")

def test_device_array_errors(cuda_device):
    ...
```

This should still skip the tests that require the pytest fixture. Add it to conftest.py to use it across test files.

Reply:
This is a nit by the way. Consider it optional or for a future enhancement if you'd like!

Reply:
Ahh that makes sense! Ignore my latest comment.